File: tests/Generator/Client/resource/typescript_collection/BarYear09Resource.ts
/**
 * BarYear09Resource generated on 0000-00-00
 * {@link https://github.com/apioo}
 */

import Axios, {AxiosInstance, AxiosPromise} from "axios";


export default class BarYear09Resource {
    private url: string;
    private token: string;
    private httpClient: AxiosInstance;

    private year: string;

    public constructor(year: string, baseUrl: string, token: string, httpClient?: AxiosInstance) {
        this.year = year;

        this.url = baseUrl + "/bar/"+year+"";
        this.token = token;
        this.httpClient = httpClient ? httpClient : Axios.create();
    }

    /**
     * Returns a collection
     *
     * @returns {AxiosPromise<EntryCollection>}
     */
    public get(): AxiosPromise<EntryCollection> {
        let params = {
            method: "GET",
        };

        return this.httpClient.get<EntryCollection>(this.url, params);
    }

    /**
     * @param {EntryCreate} data
     * @returns {AxiosPromise<EntryMessage>}
     */
    public post(data: EntryCreate): AxiosPromise<EntryMessage> {
        let params = {
            method: "POST",
        };

        return this.httpClient.post<EntryMessage>(this.url, data, params);
    }

}

File: tests/Generator/Client/resource/go_collection/bar_year09_resource.go

// BarYear09Resource generated on 0000-00-00
// {@link https://github.com/apioo}



import (
    "encoding/json"
    "io/ioutil"
    "net/http"
    "time"
)

type BarYear09Resource struct {
    BaseUrl string
    Token string
    Year string
}

// Get Returns a collection
func (r BarYear09Resource) Get() EntryCollection {


    req, err := http.NewRequest("GET", r.BaseURL + url, nil)

    client := &http.Client{}
    resp, err := client.Do(req)

    if err != nil {
        panic(err)
    }

    defer resp.Body.Close()
    respBody, _ := ioutil.ReadAll(resp.Body)

    var response EntryCollection
    json.Unmarshal(respBody, &response)

    return response
}

// Post 
func (r BarYear09Resource) Post(data EntryCreate) EntryMessage {

    raw, err := json.Marshal(data)
    if err != nil {
        panic(err)
    }
    var reqBody = bytes.NewReader(raw)

    req, err := http.NewRequest("POST", r.BaseURL + url, reqBody)
    req.Header.Set("Content-Type", "application/json")

    client := &http.Client{}
    resp, err := client.Do(req)

    if err != nil {
        panic(err)
    }

    defer resp.Body.Close()
    respBody, _ := ioutil.ReadAll(resp.Body)

    var response EntryMessage
    json.Unmarshal(respBody, &response)

    return response
}


func NewBarYear09Resource(year string, baseUrl string, token string) BarYear09Resource {
    r := BarYear09Resource {
        BaseUrl: baseUrl + "/bar/"+year+"",
        Token: token
    }
    return r
}
